pages/index.tsx

```diff
@@ -91,7 +94,7 @@ export default function Home() {
     }
 
     setHighestPriority(highestPriority);
-    setFeedArchive(storage as FeedArchiveType);
+    setFeedArchive(() => storage as FeedArchiveType);
```

Comment:
🤔 I did not find any explanation for using callback instead of actual value. Maybe you can explain?

Reply:
Sure thing:
https://stackoverflow.com/questions/42038590/when-to-use-react-setstate-callback#:~:text=so%20if%20you%20want%20to%20perform%20an%20action%20immediately%20after%20setting%20state%20on%20a%20state%20variable%20and%20then%20return%20a%20result%2C%20a%20callback%20will%20be%20useful

Reply:
Ok, but I don't understand the difference in our case `setFeedArchive(storage)` vs `setFeedArchive(() => storage)`

Reply:
I did it with the callback to make sure it happens immediately

Reply:
I had a discussion with my teammates about this and we still don't understand the difference.

> to make sure it happens immediately

both `setFeedArchive(storage)`and `setFeedArchive(() => storage)` update the state at the same time (as I understand😅).

--- 

I did a test with `setFeedArchive(storage)` and I see no difference in behavior in UX.

---

I am happy to debate this and see if there is a difference between
`setState(newState)` and `setState(() => newState)`.

Maybe we can ask on some forums?🤔

Reply:
`updateFeeds` is re-declared on every re-render. I'll wrap it in a useCallback instead then